### PR TITLE
Gh 904/settings rate app

### PR DESCRIPTION
**Pull request by DmitryBespalov**
Implements #904 and #905

### CODE REVIEW

File: Multisig/Logic/AppReviewController.swift [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    /// Trigger in-app review dialog on 3rd app start (for the first time)
+    ///      - don't count app starts triggered by a push notification
+    ///
+    /// After the dialog is triggered for the first time, use the same logic
+    /// as before for subsequent in-app reviews
+    /// (to handle cases when the user canceled the flow).
+    ///
+    /// This means count to 3 again, not including starts from push notifications.
+    func pullAppReviewTrigger() {
+        guard AppSettings.termsAccepted else { return }
```
Comment (sche) [inline Multisig/Logic/AppReviewController.swift:26]: nice :)

File: Multisig/Logic/Models/AppSettings.swift [hunk 2]
```diff
@@ -21,58 +21,49 @@ extension AppSettings {
...
+        self.path = path
     }
 
-    static var importedOwnerKey: Bool {
+    var wrappedValue: T {
         get {
             dispatchPrecondition(condition: .onQueue(.main))
-            return current().importedOwnerKey
+            let object = AppSettings.current()
+            return object[keyPath: path]
```
Comment (sche) [inline Multisig/Logic/Models/AppSettings.swift:60]: advanced magic 👍🏻

**Review by sche — APPROVED** [review]
👍🏻

File: Multisig/Cross-layer/Configuration/Config.Example.xcconfig [hunk 3]
```diff
@@ -131,6 +131,8 @@ HELP_CENTER_URL         = https:/$()/help.gnosis-safe.io/
 FEATURE_SUGGESTION_URL  = https:/$()/safe.cnflx.io/
 CONTACT_EMAIL           = safe@gnosis.io
 
+APP_STORE_REVIEW_URL = https:/$()/apps.apple.com/app/id1515759131?action=write-review
```
Comment (MouazAlzahabi) [inline Multisig/Cross-layer/Configuration/Config.Example.xcconfig]: Just wanted to note that using app store id will make this url work only for current app but not others, ie. staging Rinkeby..

you can do same thing using the bundle id using the following url formate 
https://itunes.apple.com/@StoreID/lookup?bundleId=@BundleID

Easy way to get storeID as explained here: 
https://stackoverflow.com/a/61597181/5093995

**Review by MouazAlzahabi — APPROVED** [review]

File: Multisig/Cross-layer/Configuration/Config.Example.xcconfig (hunk 3 above)
Comment (DmitryBespalov) [reply Multisig/Cross-layer/Configuration/Config.Example.xcconfig]: nice catch, will update the setting

**Pull request merged**